### PR TITLE
How to upload a custom vector map .json for Dashboards

### PR DESCRIPTION
**Pull request by alicejw-aws**
…s/794#issuecomment-1183357492

Signed-off-by: alicejw <alicejw@amazon.com>

### Description
Document how to upload a Geo .json file for a custom region map in the Dashboards.

### Issues Resolved
[794](https://github.com/opensearch-project/documentation-website/issues/794#issuecomment-1183357492)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by kolchfa-aws — APPROVED** [review]
LGTM

File: _dashboards/geojson-regionmaps.md [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+title: Region map visualizations
+nav_order: 7
+---
+
+# Region map visualizations
+
+OpenSearch Dashboards provides basic map tiles with a standard vector map that you can use to create your region map visualizations. You can configure the base map tiles using the WMS map server.
+
+You can't configure a server to support user-defined vector map layers. However, you can configure your own GeoJSON file and upload it for this purpose. 
+{: .note }
```
Comment (Shivamdhar) [inline _dashboards/geojson-regionmaps.md:12]: is this a placeholder? seems like something is missing here 😅

File: _dashboards/geojson-regionmaps.md [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+OpenSearch also provides a standard set of GeoJSON files to provide a vector map with your regional maps.
+
+## Custom vector maps with GeoJSON
+
+If you have a specific locale that is not provided by OpenSearch Dashboards vector maps, such as a US county or US zip code, you can create your own custom vector map with a GeoJSON file. To use this feature, you have to install both the [Dashboards-Maps frontend plugin](https://github.com/opensearch-project/dashboards-maps) and the [OpenSearch Geospatial backend plugin](https://github.com/opensearch-project/geospatial).
+
+GeoJSON format allows you to encode geographic data structures. To learn more about the GeoJSON specification, go to [geojson.org](https://geojson.org/).
+
+You can use this tool to extract GeoJSON files: [geojson.io](https://geojson.io/#map=2/20.0/0.0).
+{: .tip }
```
Comment (Shivamdhar) [inline _dashboards/geojson-regionmaps.md:23]: I am not able to see these macros `.tip` and `.note` in action in the preview, please check if these are syntactically correct.

File: _dashboards/geojson-regionmaps.md (hunk 1 above)
Comment (kolchfa-aws) [reply _dashboards/geojson-regionmaps.md:12]: The {: .note} and {: .tip} are style markers in Markdown. It displays correctly on the website.

File: _dashboards/geojson-regionmaps.md (hunk 2 above)
Comment (kolchfa-aws) [reply _dashboards/geojson-regionmaps.md:23]: I know it does not render correctly in GitHub, but it does on the doc website.

File: _dashboards/geojson-regionmaps.md (hunk 1 above)
Comment (Shivamdhar) [reply _dashboards/geojson-regionmaps.md:12]: Ah got it! Thanks : )

**Review by Shivamdhar — APPROVED** [review]

**Comment by kolchfa-aws** [comment]
@kgcreative and @btzeng: could one of you review this PR? Thanks!

File: _dashboards/geojson-regionmaps.md [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+You can also create your own custom vector map by uploading a JSON file that contains GEO data for your customized regional maps. The JSON file contains vector layers for visualization.
+
+1. Prepare a JSON file to upload. Make sure the file has either .geojson or .json extension.
+1. On the top menu bar, go to **OpenSearch Dashboards > Visualize**.
+1. Select the **Create Visualization** button.
+1. Select **Region Map**.
+1. Choose a source, for example: **[Flights] Flight Log**.
+1. In the right panel, select **Import Vector Map**. 
+1. In **Upload map**, select or drag and drop your JSON file.
+Enter **Map name prefix** (for example, `india-china`). Your map will have the prefix that you defined followed by the `-map` suffix (for example, `india-china-map`).
```
Comment (Shivamdhar) [inline _dashboards/geojson-regionmaps.md:34]: [nit] we can use generic name here -> `sample-map` or `county-map`.

**Review by Naarcha-AWS — APPROVED** [review]

File: _dashboards/geojson-regionmaps.md [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+1. In **Upload map**, select or drag and drop your JSON file.
+Enter **Map name prefix** (for example, `us-county`). Your map will have the prefix that you defined followed by the `-map` suffix (for example, `us-county-map`). 
+Select the **Import file** button. 
+Once the upload is successful, you will see a popup prompting you to refresh the map. Select the **Refresh** button.
+
+<img src="{{site.url}}{{site.baseurl}}/images/import-vector-map.png" alt="import a Geo .json file" width="380"/>
+
+## Layer options
+
+If you upload a custom Geo JSON file, you can toggle between the layer options: **Default vector map** or **Custom vector map**.
```
[the remark this answers is not in the document]
Comment (VijayanB) [reply _dashboards/geojson-regionmaps.md:42]: Also geoJSON is single word

[the remark this answers is not in the document]
Comment (kolchfa-aws) [reply _dashboards/geojson-regionmaps.md]: Thank you! I changed it to be one word

File: _dashboards/geojson-regionmaps.md [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+1. In **Upload map**, select or drag and drop your JSON file.
+Enter **Map name prefix** (for example, `us-county`). Your map will have the prefix that you defined followed by the `-map` suffix (for example, `us-county-map`). 
+Select the **Import file** button. 
+Once the upload is successful, you will see a popup prompting you to refresh the map. Select the **Refresh** button.
+
+<img src="{{site.url}}{{site.baseurl}}/images/import-vector-map.png" alt="import a Geo .json file" width="380"/>
+
+## Layer options
+
+If you upload a custom GeoJSON file, you can toggle between the layer options: **Default vector map** or **Custom vector map**.
```
Comment (natebower) [inline _dashboards/geojson-regionmaps.md:42]: Would it make sense to change "the" to "two"?

File: _dashboards/geojson-regionmaps.md [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+
```
Comment (natebower) [inline _dashboards/geojson-regionmaps.md:43]: I think we need a title to introduce this procedure. See https://alpha-docs-aws.amazon.com/awsstyleguide/latest/styleguide/Process_Procedure.html.

File: _dashboards/geojson-regionmaps.md [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+You can use [geojson.io](https://geojson.io/#map=2/20.0/0.0) to extract GeoJSON files.
+{: .tip}
+
+You can also create your own custom vector map by uploading a JSON file that contains GEO data for your customized regional maps. The JSON file contains vector layers for visualization.
+
+1. Prepare a JSON file to upload. Make sure the file has either .geojson or .json extension.
+1. On the top menu bar, go to **OpenSearch Dashboards > Visualize**.
+1. Select the **Create Visualization** button.
+1. Select **Region Map**.
+1. Choose a source, for example: **[Flights] Flight Log**.
```
Comment (natebower) [inline _dashboards/geojson-regionmaps.md:31]: Change to "Choose a source. For example,"

File: _dashboards/geojson-regionmaps.md [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+<img src="{{site.url}}{{site.baseurl}}/images/import-vector-map.png" alt="import a Geo .json file" width="380"/>
+
+## Layer options
+
+If you upload a custom GeoJSON file, you can toggle between the layer options: **Default vector map** or **Custom vector map**.
+
+1. On the top menu bar, go to **OpenSearch Dashboards > Visualize**.
+1. Select the **Create Visualization** button.
+1. Select **Region Map**.
+1. Choose a source, for example: **[Flights] Flight Log**.
```
Comment (natebower) [inline _dashboards/geojson-regionmaps.md:47]: Change to "Choose a source. For example,"

**Review by natebower — COMMENTED** [review]
@kolchfa-aws Please see my comments and let me know if you have any questions. Thanks!

File: _dashboards/geojson-regionmaps.md [hunk 9]
```diff
@@ -0,0 +1,70 @@
+---
+layout: default
+title: Region map visualizations
+nav_order: 7
+---
+
+# Region map visualizations
+
+OpenSearch Dashboards provides basic map tiles with a standard vector map that you can use to create your region map visualizations. You can configure the base map tiles using the WMS map server.
```
[the remark this answers is not in the document]
Comment (kolchfa-aws) [reply _dashboards/geojson-regionmaps.md:9]: Done

File: _dashboards/geojson-regionmaps.md [hunk 10]
```diff
@@ -0,0 +1,70 @@
...
+
+You can't configure a server to support user-defined vector map layers. However, you can configure your own GeoJSON file and upload it for this purpose.
```
[the remark this answers is not in the document]
Comment (kolchfa-aws) [reply _dashboards/geojson-regionmaps.md:11]: Done

File: _dashboards/geojson-regionmaps.md [hunk 11]
```diff
@@ -0,0 +1,70 @@
...
+OpenSearch Dashboards provides basic map tiles with a standard vector map that you can use to create your region map visualizations. You can configure the base map tiles using the WMS map server.
+
+You can't configure a server to support user-defined vector map layers. However, you can configure your own GeoJSON file and upload it for this purpose. 
+{: .note}
+
+OpenSearch also provides a standard set of GeoJSON files to provide a vector map with your regional maps.
+
+## Custom vector maps with GeoJSON
+
+If you have a specific locale that is not provided by OpenSearch Dashboards vector maps, such as a US county or US zip code, you can create your own custom vector map with a GeoJSON file. To use this feature, you have to install both the [Dashboards-Maps frontend plugin](https://github.com/opensearch-project/dashboards-maps) and the [OpenSearch Geospatial backend plugin](https://github.com/opensearch-project/geospatial).
```
[the remark this answers is not in the document]
Comment (kolchfa-aws) [reply _dashboards/geojson-regionmaps.md:18]: Done

File: _dashboards/geojson-regionmaps.md [hunk 12]
```diff
@@ -0,0 +1,70 @@
...
+
+GeoJSON format allows you to encode geographic data structures. To learn more about the GeoJSON specification, go to [geojson.org](https://geojson.org/).
+
+You can use [geojson.io](https://geojson.io/#map=2/20.0/0.0) to extract GeoJSON files.
+{: .tip}
+
+You can also create your own custom vector map by uploading a JSON file that contains GEO data for your customized regional maps. The JSON file contains vector layers for visualization.
+
+1. Prepare a JSON file to upload. Make sure the file has either .geojson or .json extension.
```
[the remark this answers is not in the document]
Comment (kolchfa-aws) [reply _dashboards/geojson-regionmaps.md:27]: Done

File: _dashboards/geojson-regionmaps.md (hunk 5 above)
Comment (kolchfa-aws) [reply _dashboards/geojson-regionmaps.md:42]: Done

File: _dashboards/geojson-regionmaps.md [hunk 13]
```diff
@@ -0,0 +1,70 @@
...
+
+If you have a specific locale that is not provided by OpenSearch Dashboards vector maps, such as a US county or US zip code, you can create your own custom vector map with a GeoJSON file. To use this feature, you have to install both the [Dashboards-Maps frontend plugin](https://github.com/opensearch-project/dashboards-maps) and the [OpenSearch Geospatial backend plugin](https://github.com/opensearch-project/geospatial).
+
+GeoJSON format allows you to encode geographic data structures. To learn more about the GeoJSON specification, go to [geojson.org](https://geojson.org/).
+
+You can use [geojson.io](https://geojson.io/#map=2/20.0/0.0) to extract GeoJSON files.
+{: .tip}
+
+You can also create your own custom vector map by uploading a JSON file that contains GEO data for your customized regional maps. The JSON file contains vector layers for visualization.
+
```
[the remark this answers is not in the document]
Comment (kolchfa-aws) [reply _dashboards/geojson-regionmaps.md:26]: Done

File: _dashboards/geojson-regionmaps.md [hunk 14]
```diff
@@ -0,0 +1,70 @@
...
+1. Prepare a JSON file to upload. Make sure the file has either .geojson or .json extension.
+1. On the top menu bar, go to **OpenSearch Dashboards > Visualize**.
+1. Select the **Create Visualization** button.
+1. Select **Region Map**.
+1. Choose a source, for example: **[Flights] Flight Log**.
+1. In the right panel, select **Import Vector Map**. 
+1. In **Upload map**, select or drag and drop your JSON file.
+Enter **Map name prefix** (for example, `us-county`). Your map will have the prefix that you defined followed by the `-map` suffix (for example, `us-county-map`). 
+Select the **Import file** button. 
+Once the upload is successful, you will see a popup prompting you to refresh the map. Select the **Refresh** button.
```
[the remark this answers is not in the document]
Comment (kolchfa-aws) [reply _dashboards/geojson-regionmaps.md:36]: Done

File: _dashboards/geojson-regionmaps.md (hunk 7 above)
Comment (kolchfa-aws) [reply _dashboards/geojson-regionmaps.md:31]: Done

File: _dashboards/geojson-regionmaps.md (hunk 8 above)
Comment (kolchfa-aws) [reply _dashboards/geojson-regionmaps.md:47]: Done

File: _dashboards/geojson-regionmaps.md [hunk 15]
```diff
@@ -0,0 +1,70 @@
...
+---
+
+# Region map visualizations
+
+OpenSearch Dashboards provides basic map tiles with a standard vector map that you can use to create your region map visualizations. You can configure the base map tiles using the WMS map server.
+
+You can't configure a server to support user-defined vector map layers. However, you can configure your own GeoJSON file and upload it for this purpose. 
+{: .note}
+
+OpenSearch also provides a standard set of GeoJSON files to provide a vector map with your regional maps.
```
[the remark this answers is not in the document]
Comment (kolchfa-aws) [reply _dashboards/geojson-regionmaps.md:14]: Done

File: _dashboards/geojson-regionmaps.md (hunk 6 above)
Comment (kolchfa-aws) [reply _dashboards/geojson-regionmaps.md:43]: Done

**Pull request merged**